pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -27,10 +28,129 @@ import (
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:scope=Namespaced
 type StorageCohort struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the behavior of a cohort.
+	// +required
+	Spec StorageCohortSpec `json:"spec"`
+
+	// Most recently observed status of the cohort.
+	// Populated by the cohort operator or cohort manager.
+	// +optional
+	Status StorageCohortStatus `json:"status,omitempty"`
 }
 
+// StorageCohortSpec describes the attributes that a cohort is created with.
+type StorageCohortSpec struct {
+	// NodeSelector is used to specify the cohort to be considered
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+
+	// CohortManager stores all the details about the cohort manager responsible for
+	// managing the cohort
+	// +optional
+	CohortManager interface{} `json:"cohortManager,omitempty"`
+
+	// DefaultStorageProvisioner is the default provisioner for the cohort which can be used
+	// for provisioning pools or volumes when no provisioner is specified in the storage pool
+	// For example: "openebs.io/scp-lvm-provisioner" or "openebs.io/scp-device-provisioner".
+	// +optional
+	DefaultStorageProvisioner string `json:"defaultStorageProvisioner,omitempty"`
+}
+
+// StorageCohortStatus stores information about the current status of a storage cohort.
+type StorageCohortStatus struct {
+	// Components is an array of different component's conditions which the cohort is comprised of.
+	// +optional
+	Components []ComponentStatus `json:"components,omitempty"`
```

Comment:
Can you mention the examples of different components that we are referring here as list?

Reply:
Good point @iyashu . IMO, this need not be a list now, as the structure internally contains the lists already.